pkg/controller/registry/reconciler/grpc.go

```diff
@@ -198,12 +199,20 @@ func (c *GrpcRegistryReconciler) ensureUpdatePod(source grpcCatalogSourceDecorat
 			updatePod.Labels[CatalogSourceUpdateKey] = ""
 
 			// Update the update pod to promote it to serving pod
-			_, err := c.OpClient.KubernetesInterface().CoreV1().Pods(source.GetNamespace()).Update(context.TODO(), updatePod, metav1.UpdateOptions{})
-			if err != nil {
-				return errors.Wrapf(err, "error creating new pod: %s", source.Pod().GetName())
+			forceApply := true
+			opts := metav1.PatchOptions{
+				Force:        &forceApply,
+				FieldManager: "operator-lifecycle-manager-registry-reconciler",
 			}
 
-			break
+			updatePodBytes, err := client.Apply.Data(updatePod)
+			if err != nil {
+				return errors.Wrapf(err, "error serializing updated catalog source pod to bytes: %s", source.Pod().GetName())
+			}
+			_, err = c.OpClient.KubernetesInterface().CoreV1().Pods(source.GetNamespace()).Patch(context.TODO(), updatePod.Name, types.ApplyPatchType, updatePodBytes, opts)
```

Comment:
I recently wrote a utility function that gets rid of the boilerplate for performing SSAs in our tests. See https://github.com/operator-framework/operator-lifecycle-manager/pull/1641#discussion_r455507033 for more details. PTAL.

Reply:
This is the context of the reconciler code though, not the e2e tests. Should we move the library to `pkg` somewhere and have access to it in multiple places? Rather than import a test library in the live code

Reply:
I've moved the SSA code over to a pkg/lib/util. It needs a full kube client to work, and the registry reconciler won't be able to use it without adding that in.

Reply:
Seems like if we want to use nick’s SSA library in the code we need to plumb the client in by hanging it off the Operator object at the top-level of the catalog operator

Reply:
I've added an SSA client to the `TestContext` used in our e2e tests. It should be easy to add it into the operator object if we need it there